### PR TITLE
enable markdown linter in this repository

### PR DESCRIPTION
**Pull request by dhellmann**
Use github's super-linter, since that is easy to configure with a github
action. See .github/workflows/linter.yml for the job configuration.

Add a Makefile with a command to run the linter in a container locally.

Disable a few rules that seem pedantic and that we break a lot. See
.markdownlint.yaml for details.

Fix all of the other issues, using a separate commit for each directory
or top-level enhancement.

- Wrap lines longer than 400 characters using emacs' fill-paragraph with
  fill-column set to 70.
- Remove trailing whitespace.
- Correct indentation for unordered lists.
- Add language specifiers to fenced code blocks.
- Remove trailing punctuation from headers.
- Standardize on header format using prefix only.
- Correct header increments (avoid an H3 followed by H5).
- Clean up use of emphasis markup and whitespace.

### CODE REVIEW

**Comment by dhellmann** [comment]
/cc @russellb @wking @sferich888 

Note the action won't run speculatively, but it will run on other PRs after we merge this one.

File: enhancements/cloud-integration/aws/aws-sts-support.md [hunk 1]
```diff
@@ -87,15 +101,26 @@ All CVO managed operators will also now mount in a [Kubernetes Projected Service
...
+iterate:
 
   1. Create an IAM Role with no policy attached, type = web identity, identity provider created by the admin manually.
     1. If the role will not be shared across multiple clusters and should be removed when the cluster is deprovisioned, tag it with `kubernetes.io/cluster/[infraid]=owned`.
     1. Note that openshift-install destroy cluster will not cleanup the OIDC provider itself today.
   1. Attach an in-line policy to the role which matches the permissions in the `CredentialsRequest` or documentation.
   1. Ensure the trust relationship for the Role points to the cluster’s AWS IAM OIDC ARN with conditions to limit which `ServiceAccount` can assume this role.
-     ```json
+
+```json
```
Comment (wking) [inline enhancements/cloud-integration/aws/aws-sts-support.md:123]: this was probably intended to be indented to nest under the `Ensure the trust relationship...` point.

**Comment by wking** [comment]
/lgtm

This is so big, and will be a pain to rebase.  I'd rather land it quickly, even if we end up accidentally un-optimizing something like [list nesting][1].  We can always polish in follow-up PRs if it seems like it's worth it.

[1]: https://github.com/openshift/enhancements/pull/594#discussion_r560532199

**Pull request merged**